atlp-rwanda/destructors-ec-bn
# feat(product availability): mark product as available/unavailable

What does this PR do?

A seller should be able to mark product as available/unavailable
A seller should be able to update product from their seller's collection
A seller should be able to delete specific product from their seller's collection 

Description of Task to be completed?

 A seller should mark product as available/unavailable as the owner of the product
A seller should delete  product as the owner of the product
A seller should update   product  as the owner of the product

How should this be manually tested?
Installation
1. Clone the repository by running

git clone https://github.com/atlp-rwanda/destructors-ec-bn.git
2. Navigate to the project directory by running cd destructors-ec-bn.

cd destructors-ec-bn
 cd  ft-product-availability-184637144
3. Install the necessary dependencies by running npm install.

npm install
4. Run the development server by running npm run dev. This will start the server using Nodemon, which will automatically reload the server when changes are made to the code.

npm run dev
5. Run the tests by running npm run test

npm run test
6. Held on swagger docs on http://localhost:<port>/api/v1/api-docs.

Signup into the app and get token and insert them into the Authorization and make sure that the value is started by Bearer.
PUTCH/api/v1/products/{id}/availability: mark availability product
PUTCH /api/v1/products/{id}:update specific product from  thier sellers's collection
DELETE/api/v1/products/{id}:delete product from thier sellers's collection

Any background context you want to provide?
    None
What are the relevant pivotal tracker/Trello stories?
      Tracker:184637144
Screenshots (if appropriate)
none
Questions:
